fairscale/optim/oss.py

```diff
@@ -166,19 +167,22 @@ def step(self, closure: Optional[Callable[[], float]] = None, **kwargs: Any) ->
         OSS._sync_param_groups(self.param_groups, self.optim.param_groups)
 
         # Catch a possible change of devices in between OSS construction and step()
-        if self._default_device.type != self.param_groups[0]["params"][0].device.type:
-            logging.info("OSS detected that the parameter changed devices, re-allocating buffers")
-            self._clear_cache()
-            self.refresh_trainable()
+        with profiler.record_function("oss_refresh_trainable"):
```

Comment:
Do you want to add the fairscale namespace similar to [offload](https://github.com/facebookresearch/fairscale/blob/5cddaea485e9d8aa46f4e5ed440d2677939ae55f/fairscale/experimental/nn/offload.py#L168)? I think it will be clearer to the user that this is an internal record_function call.

Reply:
that was the idea behind "oss", kind of a namespace. I actually think that a long name is not a good idea in a trace, it will probably be unreadable

Reply:
This is a typical trace, I would use the smallest possible label to try to keep it readable

![Screen Shot 2021-04-22 at 5 10 24 PM](https://user-images.githubusercontent.com/3072861/116123271-ad13b100-a677-11eb-866c-149dafdfe948.png)

Reply:
we probably want see the entire label anyway unless you expand the trace and once you do you will see the entire label(tested this with offload). My suggestion is to  standardize how we want to add these labels within FairScale in way that the user is able to differentiate between their own record_function labels and the ones we add.

Reply:
I'm personally not sold on the added value of `fairscale.experimental.nn.`, people looking into a trace probably already have an idea of what they're looking at. It does make sure that the blocks are not identifiable from a distance, same in the above trace (the`xformers` labels on some blocks are not that useful in my view), requiring people to dig in to get a readable label does not look like a usability win ?

Reply:
For instance, the aten labels above just have the "aten::name" format, and I think that works well